### ISSUE TITLE
Tracking workflow header affine issue & fix

### ISSUE THREAD

**Issue by sitek** [nipy/dipy#1832]
## Description
Using `LocalFiberTrackingPAMFlow` in 0.16, I found that streamlines were in an incorrect space relative to the input DWI files. (see first post with image on neurostars issue [here](https://neurostars.org/t/dipy-streamlines-not-aligning-with-image-space/3889))

I resolved the issue by adding a `header` to the `nib.streamlines.save()` call: 

```
from nibabel.orientations import aff2axcodes
from nibabel.streamlines import Field
import nibabel as nib

# import fa image (or other DWI)
img = nib.load(stopping_path)

# header inputs
affine = img.affine
vox_size = img.header.get_zooms()[:3]
fov_shape = img.get_fdata().shape[:3]

# create header
hdr = {}
hdr[Field.VOXEL_TO_RASMM] = affine.copy()
hdr[Field.VOXEL_SIZES] = vox_size
hdr[Field.DIMENSIONS] = fov_shape
hdr[Field.VOXEL_ORDER] = "".join(aff2axcodes(affine))

# create and save tractogram object with the header
tractogram = Tractogram(streamlines_generator, affine_to_rasmm=np.eye(4))
save(trk_file, 'tractogram_correct_affine.trk', header=hdr)
```
**note that the `Tractogram` object is created with a _blank_ affine (`np.eye(4)`) but the `header` takes the _image_ affine

Adding this or similar code [here](https://github.com/nipy/dipy/blob/8379bbde5bfe07ed2d6fa48f4f08953b298f62f5/dipy/workflows/tracking.py#L80-L102) could resolve this issue. I can make a pull request if this seems like an optimal solution, but I'm open to other solutions as well (assuming other people have the same issue).

**Comment by skoudoro** [comment]
Hi @sitek,

Thank for the report, it seems to be related to PR #1812 and issue #1722. We are in a redesign process. Feel free to give your opinion on #1812

**Comment by skoudoro** [comment]
fixed by #1812, closing
